### DISCUSSION TITLE
Can't make sense of trip.fuel_consumed

### DISCUSSION THREAD

**Discussion by masve64** [DurgNomis-drol/mytoyota#322] [Q&A]
Hi, thanks for a great work :-)

I've been fetching data from Toyota, and doing so looking at the trip.fuel_consumed data, metric=True, on a trip, I'm presented with following information:

**Km/Liter**
Trip start time: 2024-02-04 10:04:40+00:00 - (55.849, 12.4852)
Trip end time: 2024-02-04 10:25:14+00:00 - (55.8413, 12.2868)
**Fuel consumed. 0.248000**
Ev dist: 4.899000
Ev duration: 0:08:29
Dist: 16.602000

However if I choose metric=False, I get the following information:

**Miles/Gallon**
Trip start time: 2024-02-04 10:04:40+00:00 - (55.849, 12.4852)
Trip end time: 2024-02-04 10:25:14+00:00 - (55.8413, 12.2868)
**Fuel consumed. 0.248000**
Ev dist: 3.044000
Ev duration: 0:08:29
Dist: 10.316000

According to my car my fuel consumption is 14,7km/liter, and according to my Toyota APP  it's shows 7L/100km (just about the same)

So how do I get from fuel_consumed = 0.248000 liter to something that match what my Car or the APP data ?

Looking in the class trip code, fuel_consumed:

    @property
    def fuel_consumed(self) -> float:
        """The amount of fuel consumed.

        Returns
        -------
            float: The fuel consumed in liters if metric or gallons
        """
        if self._trip.summary.fuel_consumption:
            return (
                round(self._trip.summary.fuel_consumption / 4546.0, 3)
                if self._metric
                else (self._trip.summary.fuel_consumption / 1000.0)
            )

        return 0.0

However looking in class summary, fuel_consumed is calculated diffrently:

    @property
    def fuel_consumed(self) -> float:
        """The total amount of fuel consumed.

        Returns
        -------
            float: The total amount of fuel consumed in liters if metric or gallons
        """
        if self._summary.fuel_consumption:
            return (
                round(self._summary.fuel_consumption / 1000.0, 3)
                if self._metric
                else round(self._summary.fuel_consumption / 3785.0, 3)
            )

        return 0.0

**Comment by CM000n** [comment]
Hi @masve64. Looks like we have a discrepancy and the formula for converting mililiters to gallons in the trips model is currently incorrect. Should be fixed in next version: https://github.com/DurgNomis-drol/mytoyota/pull/327 Thx for reporting it 😊 

But the way you describe it, you don't want the total fuel consumed (`fuel_consumed`) but the average fuel consumption (`average_fuel_consumed`): https://github.com/DurgNomis-drol/mytoyota/blob/master/mytoyota/models/summary.py#L168

**Reply by masve64** [reply]
Thanks, if for metric=True I can assume that the fuel consumed is for the entire trip, then 0,248 Liter consumed for 16,6 km => 66,9km/liter, it did not make any sense (evenhough it's a RAV4 Hybrid) ;-)

**Reply by CM000n** [reply]
As I said before, the formula in the trips model is currently wrong 😉 
https://github.com/DurgNomis-drol/mytoyota/pull/327
